src/sch/eager.jl

```diff
@@ -3,9 +3,43 @@ const EAGER_THUNK_CHAN = Channel(typemax(Int))
 const EAGER_ID_MAP = Dict{UInt64,Int}()
 const EAGER_CONTEXT = Ref{Context}()
 const EAGER_STATE = Ref{ComputeState}()
+const EAGER_SCH_OPTS = Ref{SchedulerOptions}(SchedulerOptions(;allow_errors=true))
 
 eager_context() = isassigned(EAGER_CONTEXT) ? EAGER_CONTEXT[] : nothing
 
+"""
+    global_eager_scheduler_options(; kwargs...)
+
+Set / retrieve the global eager scheduler options.
+If no `kwargs` are passed, the current `SchedulerOptions` is returned,
+otherwise the global eager scheduler options are set to
+`SchedulerOptions(;kwargs...)` and returned.
+# Keywords
+- `kwargs...`: SchedulerOptions keywords to be set
+# Returns
+- `SchedulerOptions`: The global options used for eager scheduling.
+"""
+function global_eager_scheduler_options(; kwargs...)
+    if !isassigned(EAGER_SCH_OPTS) || !isempty(kwargs)
+        EAGER_SCH_OPTS[] = SchedulerOptions(; kwargs...)
```

Comment:
Instead of wholesale replacing it, maybe we could do an in-place modification? It feels annoying to have to specify everything again if you just want to set one option.

Thoughts?

Reply:
I haven't fully grokked how the eager scheduling works yet, but if some thunks that got pushed to `EAGER_THUNK_CHAN` have not been processed yet, would a later call to the in-place version of `eager_options`  not change the scheduler options used for those thunks retroactively? Seems like previously pushed thunks should use whatever options were in effect when they were pushed.

Reply:
Basically, we use whatever options are set whenever we access the options struct. I've left it like this because I didn't want to have to implement the logic to support what you're proposing (and the tests to make sure it works correctly). If you want to implement your proposed semantics, feel free to file a PR; otherwise, go ahead and file an issue and I'll get to it at some point in the future :slightly_smiling_face:

Reply:
Oh ok, that's fine, the current semantics probably cover 99% of use-cases, so guess it's easiest to just stick with them.

Reply:
Sorry, I was entirely wrong about the in-place modification - `SchedulerOptions` is immutable, and I think it's reasonable for it to stay like that. Marking this as resolved.